plasma_framework/test/src/exits/payment/PaymentChallengeIFEInputSpent.test.js

```diff
@@ -0,0 +1,357 @@
+        function buildInputTx() {
+            const tx = createInputTransaction(
+                [buildUtxoPos(2000, 4, 3), buildUtxoPos(2000, 10, 2)],
+                inputOwner,
+                334455,
+            );
+
+            const txBytes = web3.utils.bytesToHex(tx.rlpEncoded());
+            const outputIndex = 0;
+
+            return {
+                tx,
+                txBytes,
+                outputIndex,
+                utxoPos: buildUtxoPos(3000, 5, outputIndex),
+            };
+        }
+
+        // Sets up an IFE tx using inputTx as an input, starts the IFE and piggybacks on the input.
+        const buildPiggybackInputData = async (inputTx) => {
+            const outputAmount = 997;
+
+            const firstInput = createInputTransaction([buildUtxoPos(BLOCK_NUMBER, 3, 0)], outputOwner, 334455);
+            const firstInputUtxoPos = buildUtxoPos(3000, 66, 0);
+
+            const inFlightTx = createInFlightTx(
+                [firstInput, inputTx.tx],
+                [firstInputUtxoPos, inputTx.utxoPos],
+                alice,
+                outputAmount,
+            );
+            const rlpInFlightTxBytes = web3.utils.bytesToHex(inFlightTx.rlpEncoded());
+
+            const emptyWithdrawData = {
+                outputId: web3.utils.sha3('dummy output id'),
+                outputGuard: web3.utils.sha3('dummy output guard'),
+                exitTarget: constants.ZERO_ADDRESS,
+                token: constants.ZERO_ADDRESS,
+                amount: 0,
+            };
+
+            const inFlightExitData = {
+                exitStartTimestamp: (await time.latest()).toNumber(),
+                exitMap: 0,
+                position: INFLIGHT_EXIT_YOUNGEST_INPUT_POSITION,
+                bondOwner: alice,
+                oldestCompetitorPosition: 0,
+                inputs: [{
+                    outputId: web3.utils.sha3('dummy output id'),
+                    outputGuard: web3.utils.sha3('dummy output guard'),
+                    exitTarget: inputOwner,
+                    token: ETH,
+                    amount: 999,
+                }, {
+                    outputId: getOutputId(inputTx.txBytes, inputTx.utxoPos),
+                    outputGuard: web3.utils.sha3('dummy output guard'),
+                    exitTarget: inputOwner,
+                    token: ETH,
+                    amount: 998,
+                }, emptyWithdrawData, emptyWithdrawData],
+                outputs: [{
+                    outputId: web3.utils.sha3('dummy output id'),
+                    outputGuard: web3.utils.sha3('dummy output guard'),
+                    exitTarget: outputOwner,
+                    token: ETH,
+                    amount: outputAmount,
+                }, emptyWithdrawData, emptyWithdrawData, emptyWithdrawData],
+            };
+
+            const exitId = await this.exitIdHelper.getInFlightExitId(rlpInFlightTxBytes);
+
+            const argsInputOne = {
```

Comment:
I would name this `piggybackArgsInputOne` and same apply for another as piggyback is not the main testing target in this test file. However, I think we can even skip the piggyback part and manipulate the `exitMap` directly in test to mimic the state after piggyback directly IMO, which might/might not?  simpiflies the tests a bit.

Reply:
Good idea, it does simplify the test a bit